### PR TITLE
feat(replays): track analytics for replay search events

### PR DESCRIPTION
**Pull request by eliashussary**
## Summary
Fires a `replay.search` event `onSearch` and passes in the search field keys.

Relates to: https://github.com/getsentry/team-replay/issues/53

### CODE REVIEW

File: static/app/views/replays/replaySearchBar.tsx [hunk 1]
```diff
@@ -121,6 +123,20 @@ function ReplaySearchBar(props: Props) {
...
       fieldDefinitionGetter={getReplayFieldDefinition}
+      onSearch={(query: string) => {
+        props?.onSearch?.(query);
+        const conditions = new MutableSearch(query);
+        const searchKeys = conditions.tokens
+          .map(({key}) => key)
+          .filter(v => typeof v === 'string') as string[];
+
+        if (searchKeys.length > 0) {
+          trackAdvancedAnalyticsEvent('replay.search', {
```
Comment (billyvg) [inline static/app/views/replays/replaySearchBar.tsx:134]: This tracks to amplitude?

Comment (scefali) [reply static/app/views/replays/replaySearchBar.tsx:134]: @eliashussary @billyvg No it won't, use a comma delineated string instead here

File: static/app/utils/analytics/replayAnalyticsEvents.tsx [hunk 2]
```diff
@@ -62,6 +65,7 @@ export const replayEventMap: Record<ReplayEventKey, string | null> = {
   'replay.details-viewed': 'Viewed Replay Details',
   'replay.list-paginated': 'Paginated Replay List',
   'replay.list-sorted': 'Sorted Replay List',
+  'replay.search': 'Searched Replay',
```
Comment (ryan953) [inline static/app/utils/analytics/replayAnalyticsEvents.tsx:68]: can you alpha sort these lines please

File: static/app/views/replays/replaySearchBar.tsx [hunk 3]
```diff
@@ -121,6 +123,20 @@ function ReplaySearchBar(props: Props) {
       maxMenuHeight={500}
       hasRecentSearches
       fieldDefinitionGetter={getReplayFieldDefinition}
+      onSearch={(query: string) => {
+        props?.onSearch?.(query);
```
Comment (ryan953) [inline static/app/views/replays/replaySearchBar.tsx:127]: props shouldn't need the `?`

```suggestion
        props.onSearch?.(query);
```

File: static/app/utils/analytics/replayAnalyticsEvents.tsx [hunk 4]
```diff
@@ -46,6 +46,9 @@ export type ReplayEventParameters = {
      */
     scale_bucket: 0 | 10 | 20 | 30 | 40 | 50 | 60 | 70 | 80 | 90 | 100;
   };
+  'replay.search': {
+    search_keys: string[];
```
Comment (ryan953) [inline static/app/utils/analytics/replayAnalyticsEvents.tsx:50]: to match with @scefali's comment below. should make TS happy

```suggestion
    search_keys: string;
```

**Review by ryan953 — APPROVED** [review]

File: static/app/views/replays/replaySearchBar.tsx [hunk 5]
```diff
@@ -121,6 +123,20 @@ function ReplaySearchBar(props: Props) {
...
+        const conditions = new MutableSearch(query);
+        const searchKeys = conditions.tokens
+          .map(({key}) => key)
+          .filter(v => typeof v === 'string') as string[];
```
[the remark this answers is not in the document]
Comment (eliashussary) [reply static/app/views/replays/replaySearchBar.tsx:131]: it was an attempt to get typescript to play nicely, but as you can see by the `as string[]` it didn't work so well :)

**Pull request merged**